doc/util/litchapel.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+from __future__ import print_function
+
+import sys
+
+def titlecomment(line):
+    """Condition for a line to be a title comment"""
+    return line.startswith('//') and len(line.lstrip('//').strip()) > 0
+
+def to_pieces(handle, islearnChapelInYMinutes=False):
+    output = []
+    current_type = ''
+    current_content = []
+
+    def flush():
+        nonlocal current_content, current_type
+        if len(current_content) != 0 and any(s != '' for s in current_content):
+            while current_content[0] == '': current_content.pop(0)
+            while current_content[-1] == '': current_content.pop(-1)
+            output.append((current_type, current_content))
+            current_content = []
+
+
+    def switch_type(new_type):
+        nonlocal current_type
+        if current_type == new_type:
+            return
+        flush()
+        current_type = new_type
+
+    def push_line(line):
+        nonlocal current_content
+        current_content.append(line)
+
+    commentdepth = 0
+    state = ''
+    indentation = -1
+    chunk_index = 0
+    foundCommentExample = False
+
+    # Each line is rst or code-block
```

Comment:
As this script no longer contains rst-specific code, all references to rst should be changed to something inclusive of both kinds of output. (I'm not sure what that would be)